src/tokens/selectors.ts

```diff
@@ -499,10 +517,18 @@ export const feeCurrenciesWithPositiveBalancesSelector = createSelector(
 export const visualizeNFTsEnabledInHomeAssetsPageSelector = (state: RootState) =>
   state.app.visualizeNFTsEnabledInHomeAssetsPage
 
-export const importedTokensSelector = createSelector([tokensListSelector], (tokenList) => {
-  if (!getFeatureGate(StatsigFeatureGates.SHOW_IMPORT_TOKENS_FLOW)) {
-    return []
-  }
+export const importedTokensSelector = createSelector(
+  [tokensListSelector, networksIconSelector],
+  (tokenList, networksIcon) => {
+    if (!getFeatureGate(StatsigFeatureGates.SHOW_IMPORT_TOKENS_FLOW)) {
+      return []
+    }
 
-  return tokenList.filter((token) => token?.isManuallyImported) as TokenBalance[]
-})
+    return tokenList
+      .filter((token) => token?.isManuallyImported)
+      .map((token) => ({
+        ...token,
+        networkIconUrl: networksIcon[token.networkId] ?? token.networkIconUrl,
```

Comment:
This is to update the network icon in case we change it?
I wasn't sure if I should add this, but without this change the network icon will be fixed to the one that was calculated when importing the token, maybe we are just fine with that.

Reply:
i agree with that - then can we remove the logic for saving the imported token with a networkIconUrl? it seems like we create this url in the import token screen, save it to redux, and then overwrite this value every time?

Reply:
I know the logic ended up a bit weird, we need to store it in redux so if when we show the token list the network icon is already there. If not, it will only be shown after a fetching the balances (similar to what happens with the balance, we store the first value we have but we update it in every fetch).
I guess the right solution would be to refactor the token list selector to call the imported tokens selector first?

Reply:
i strongly think that we should not build new features and settle for "this looks weird and it's okay" - i think what you're describing is that the token import screen needs to set the network icon url because to render the asset list, this value needs to be there. so the logic that we've got here in the importTokensSelector that "refreshes" the network icon url is not used for rendering the token on the UI - so do we need this logic? i see importTokensSelector is only used in the fetch balances saga, so i think we don't need the network icon url logic there - or if we need the logic, we should update the token list selector to use this logic for imported tokens? it seems like a small detail that we're likely to forget about it later, maybe i'm missing something but i seems like the logic we have doesn't work for the concerns